ghc-lib/working-on-ghc-lib.md

```diff
@@ -183,20 +194,48 @@ git checkout da-master && git rebase master
 git checkout -t origin/da-unit-ids && git rebase master
 ```
 Obviously, you will need to deal with any rebase conflicts that come up (hopefully not often). You can test `ghc-lib` after rebasing by following the [build procedure](#how-to-build-ghc-lib-from-the-da-ghc-fork) replacing the line
-```
+```bash
 git remote add upstream git@github.com:digital-asset/ghc.git
 ```
 with
-```
+```bash
 git remote add upstream $HOME/tmp/ghc
 ```
 and then the [test procedure](#how-to-test-ghc-lib).
 
 When you are satisfied that the tests pass, you can push the changes to origin with these commands:
-```
+```bash
 cd ~/tmp/ghc
 git push origin master:master
 git push -f origin da-master:da-master
 git push -f origin da-unit-ids:da-unit-ids
 ```
 After this, release the updated `ghc-lib` following the usual [deployment procedure](#how-to-deploy-ghc-lib).
+
+## How to develop `ghc-lib`
+
+The following procedure sets up a new feature branch with starting point `da-master`.
+```bash
+mkdir ~/tmp && cd ~/tmp
+git clone https://gitlab.haskell.org/ghc/ghc.git ghc.git
+cd ghc.git
+git remote add upstream git@github.com:digital-asset/ghc.git
+git fetch upstream da-master
+git checkout -b da-master --track upstream/da-master
+git checkout -b feature-xxx da-master
```

Comment:
`git checkout -b feature-xxx --track upstream/da-master` should do the same thing in one command.

Reply:
You'd think so but in fact it creates a tracking branch.
```
Branch 'feature-xxx' set up to track remote branch 'da-master' from 'upstream'.
Switched to a new branch 'feature-xxx'
```
I'll change this to 
```
git checkout -t upstream/da-master
git checkout -b feature-xxx da-master
```
for clarity.